From: Ben Hutchings <ben@decadent.org.uk>
Date: Fri, 09 Dec 2016 22:50:13 +0000
Subject: SCSI: Fix ABI changes in 3.16.37, 3.16.49
Forwarded: not-needed

Commit f05795d3d771 ("scsi: Add intermediate STARGET_REMOVE state to
scsi_target_state") added an enumerator to enum scsi_target_state, and
renumbered the existing STARGET_DEL.  Commit f9279c968c25 ("scsi: Add
STARGET_CREATED_REMOVE state to scsi_target_state") did the same
again.

The target state doesn't appear to be checked anywhere outside of the
SCSI core, but just in case it is reorder the enumerators to maintain
the value of STARGET_DEL.

Hide the change from genksyms.

---
--- a/include/scsi/scsi_device.h
+++ b/include/scsi/scsi_device.h
@@ -264,9 +264,11 @@ struct scsi_dh_data {
 enum scsi_target_state {
 	STARGET_CREATED = 1,
 	STARGET_RUNNING,
+	STARGET_DEL,
+#ifndef __GENKSYMS__
 	STARGET_REMOVE,
 	STARGET_CREATED_REMOVE,
-	STARGET_DEL,
+#endif
 };
 
 /*
